src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,180 @@
+    // if the user released on a different target, cancel!
+    if (this.startEl !== e.target) return;
+    const direction = this.resolveDirection(e);
+
+    if (direction) {
+      const eventDetail: SwipeEvent = {
+        direction,
+        distanceX: Math.abs(this.xDown - e.clientX),
+        distanceY: Math.abs(this.yDown - e.clientY),
+        touchStartOriginalEvent: this.startEvent,
+        touchEndOriginalEvent: e
+      };
+
+      // fire `swiped` event on the element that started the swipe
+      const event = new CustomEvent('swipe', {bubbles: true, cancelable: true, detail: eventDetail});
+      this.dispatchEvent(overrideEvent(event, 'target', this.target));
+
+      // fire `swiped:${dir}` event on the element that started the swipe
+      const dirEvent = new CustomEvent(`swipe:${direction}`, {bubbles: true, cancelable: true, detail: eventDetail});
+      this.dispatchEvent(overrideEvent(dirEvent, 'target', this.target));
+    }
+
+    // reset values
+    this.startEl = null;
+  }
+
+  /**
+   * @returns threshold based on number and units
+   */
+  protected resolveSwipeThreshold(): number {
+    let swipeThreshold = this.config.threshold;
+
+    if (this.config.units === 'vh') {
+      swipeThreshold = Math.round((swipeThreshold / 100) * document.documentElement.clientHeight); // get percentage of viewport height in pixels
+    }
+    if (this.config.units === 'vw') {
+      swipeThreshold = Math.round((swipeThreshold / 100) * document.documentElement.clientWidth); // get percentage of viewport width in pixels
+    }
+
+    return swipeThreshold;
+  }
+
+  /**
+   * Returns swipe direction based on distance between swipe start and end points
+   * @param e - pointer event
+   * @returns direction of swipe {@link SwipeDirection}
+   */
+  protected resolveDirection(e: PointerEvent): SwipeDirection {
+    const xDiff = this.xDown - e.clientX;
+    const yDiff = this.yDown - e.clientY;
+    const swipeThreshold = this.resolveSwipeThreshold();
+    const timeDiff = e.timeStamp - this.timeDown;
+
+    if (Math.abs(xDiff) > Math.abs(yDiff) && Math.abs(xDiff) > swipeThreshold && timeDiff < this.config.timeout) {
+      return xDiff > 0 ? 'left' : 'right';
+    }
+    if (Math.abs(yDiff) > swipeThreshold && timeDiff < this.config.timeout) {
+      return yDiff > 0 ? 'up' : 'down';
+    }
+
+    return 'left';
+  }
+
+  /** Unsubscribes from the observed target {@link Element} changes */
+  public override removeEventListener(callback: EventListener): void;
+  public override removeEventListener(event: SwipeEventName, callback: EventListener): void;
+  public override removeEventListener(event: any, callback: EventListener = event): void {
+    super.removeEventListener(event, callback);
+    if (this.getEventListeners().length > 0) return;
+
+    this.target.removeEventListener('pointerdown', this.handleStart.bind(this), false);
+    this.target.removeEventListener('pointerup', this.handleEnd.bind(this), false);
```

Comment:
```suggestion
    this.target.removeEventListener('pointerdown', this.handleStart.bind(this), false);
    this.target.removeEventListener('pointerup', this.handleEnd.bind(this), false);
```
Incorrect the subscription will not happen.
According to the `bind` method spec, it returns a NEW (every time) function that freezes the original function call context to the passed object.
This new function(s) is not subscribed to the element, so it can not be unsubscribed.
Consider @bind decorator, constructor binding, or better use internal ESLEventListener.subscribe API to create internal subscription using the ESL Event Utils module capabilities